src/scripts.js

```diff
@@ -0,0 +1,225 @@
+},
+
+  blockSource: event => {
+    const host = linkElement.hostname
+    console.log(host)
+  }
+
+}
+
+function assignElement(elementType, elementClassName, elementTextContent){
+  const output = document.createElement(elementType)
+  output.className = elementClassName
+  output.textContent = elementTextContent
+  return output
+}
+
+
+//creates each story panel from fetch response
+function createStoryPanel(article){
+  const storyDivElement = assignElement("div", "story-div-element")
+  const headerElement = assignElement("div", "story-header")
+  const headerLeftElement = assignElement("div", "story-header-left")
+  const publicationInfoElement = assignElement("span", "publication-info")
+  const headlineElement = assignElement("h2", "headline", article.title)
+  const publicationNameElement = assignElement("h4", "publication-name", article.source.name)
+  const publicationDateElement = assignElement("h4", "publication-date", convertDateForDisplay(article.publishedAt))
+  const blockButtonElement = assignElement("button", "block-button", "Block")
+  const storyImageElement = assignElement("img", "story-image")
+  storyImageElement.setAttribute("src", article.urlToImage)
+  if (storyImageElement.src === "http://127.0.0.1:3000/null"){
+    storyImageElement.setAttribute("src", "../images/No_Image_Available.jpg")
+  }
+  const descriptionElement = assignElement("p", "description-element", article.description)
+  const linkElement = assignElement("a", "full-story-link", "See full story")
+  linkElement.setAttribute("href", article.url)
+  blockButtonElement.addEventListener("click", event => {
+    let host = linkElement.hostname
+    host = host.replace("www.", "")
+    apiRequests.blockList.push(host)
+    const blockedDomains = apiRequests.blockList.join(",")
+    apiRequests.updateURL("excludeDomains", blockedDomains)
+    fetchNews(apiRequests.getURL())
+  })
+
+  publicationInfoElement.appendChild(publicationNameElement)
+  publicationInfoElement.appendChild(publicationDateElement)
+
+  headerLeftElement.appendChild(headlineElement)
+  headerLeftElement.appendChild(publicationInfoElement)
+  headerLeftElement.appendChild(blockButtonElement)
+
+  headerElement.appendChild(headerLeftElement)
+  headerElement.appendChild(storyImageElement)
+
+  storyDivElement.appendChild(headerElement)
+  storyDivElement.appendChild(descriptionElement)
+  storyDivElement.appendChild(linkElement)
+  newsDisplayElement.appendChild(storyDivElement)
+}
+
+
+
+//formats date for search
+function formatDate(monthsAgo){
+  if ( monthsAgo != "all-time") {
+    const newDate = new Date
+    newDate.setMonth(newDate.getMonth() - monthsAgo);
+    const isoDate = newDate.toISOString().slice(0,10)
+    return `${isoDate}`
+  }else{
+    return null
+  }
+}
+
+//formats date for display
+
+function convertDateForDisplay(date){
+  const inputDate = new Date(date)
+  return inputDate.toLocaleString()
+}
+
+
+function fetchNews(apiAddress){
+  fetch(apiAddress)
```

Comment:
nice. good to see how you pass in url as param and implement display logic in own function